pkg/controller/controller.go

```diff
@@ -248,30 +210,16 @@ func getDriverCapabilities(conn *grpc.ClientConn, timeout time.Duration) (sets.I
 	}
 
 	capabilities := make(sets.Int)
-	for _, cap := range pluginCaps {
-		if cap == nil {
-			continue
-		}
-		service := cap.GetService()
-		if service == nil {
-			continue
-		}
-		switch service.GetType() {
+	for cap := range pluginCaps {
+		switch cap {
```

Comment:
Is mixing plugin and controller capabilities into the same set correct?  Could you have conflicts in values?

Reply:
`PluginCapability_*` and `ControllerCapability_*` are part of the same enum defined in the provisioner, they don't come from CSI:

https://github.com/kubernetes-csi/external-provisioner/blob/e63995ffb48736dc6275b5e8702e2dc8868c57ef/pkg/controller/controller.go#L160-L165

Reply:
Thanks for pointing that out. Is there a reason why it needs to have one more level of translation from CSI capabilities?

Reply:
Can we sort it out in a separate PR? I'd like to cache the capabilities so provisioner does not load them from the driver every time it needs them. Refactoring of the enum would be better there.

Reply:
Sure do you want to use csi-lib-utils 0.4.0-rc1 here?

Reply:
Updated to 0.4.0-rc1